ssz/sedes/bitlist.py

```diff
@@ -0,0 +1,80 @@
+from typing import (
+    Sequence,
+    Union,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes.base import (
+    BaseCompositeSedes,
+)
+from ssz.utils import (
+    merkleize,
+    mix_in_length,
+    pack_bitvector_bitlist,
+)
+
+BytesOrByteArray = Union[bytes, bytearray]
+
+
+class Bitlist(BaseCompositeSedes[BytesOrByteArray, bytes]):
+    def __init__(self, bit_count: int) -> None:
+        if bit_count < 0:
+            raise TypeError("Max bit count cannot be negative")
+        self.bit_count = bit_count
+
+    #
+    # Size
+    #
+    is_fixed_sized = False
+
+    def get_fixed_size(self):
+        raise ValueError("byte list has no static size")
+
+    #
+    # Serialization
+    #
+    def serialize(self, value: BytesOrByteArray) -> bytes:
+        len_value = len(value)
+        if len_value > self.bit_count:
+            raise SerializationError(
+                f"Cannot serialize length {len_value} bit array as Bitlist[{self.bit_count}]"
+            )
+
+        if len_value == 0:
+            return b'\x01'
+
+        as_bytearray = [0] * (len_value // 8 + 1)
+        for i in range(len_value):
+            as_bytearray[i // 8] |= value[i] << (i % 8)
+        as_bytearray[len_value // 8] |= 1 << (len_value % 8)
+        return bytes(as_bytearray)
```

Comment:
I think we may be able to avoid the mutability and generally make it a little easier to read with something like this (untested, and not entirely certain if I understood everything correctly):

```Py
@apply_to_return_value(bytes)  # from eth_utils
def serialize(self, value):
    terminated_bits = tuple(value) + (True,)
    for bits_in_byte in partition(8, value, pad=False):
        yield sum(int(bit_value) << bit_index for bit_index, bit_value in enumerate(bits_in_byte))
```